rmgpy/data/kinetics/database.py

```diff
@@ -107,86 +107,101 @@ def load(self, path, families=None, libraries=None, depositories=None):
 
     def loadRecommendedFamiliesList(self, filepath):
         """
-        Load the list of recommended families from the given file
-        
-        The file is usually 'kinetics/families/recommended.py'.
-        This is stored as a dictionary of True or False values (checked here),
-        and should contain entries for every available family (checked in loadFamilies).
+        Load the recommended families from the given file.
+        The file is usually stored at 'kinetics/families/recommended.py'.
+
+        The old style was as a dictionary named `recommendedFamilies`
+        containing all family names as keys with True/False values.
+
+        The new style is as multiple sets with unique names which can be
+        used individually or in combination.
+
+        Both styles can be loaded by this method.
         """
-        try:
-            global_context = {}
-            global_context['__builtins__'] = None
-            global_context['True'] = True
-            global_context['False'] = False
-            local_context = {}
-            local_context['__builtins__'] = None
-            f = open(filepath, 'r')
-            exec f in global_context, local_context
-            f.close()
-            self.recommendedFamilies = local_context['recommendedFamilies']
-        except Exception, e:
-            raise DatabaseError('Error while reading list of recommended families from {0}/recommended.py.\n{1}'.format(filepath,e))
-        for recommended in self.recommendedFamilies.values():
-            if not isinstance(recommended, bool):
-                raise DatabaseError("recommendedFamilies dictionary should contain only True or False values")
+        import imp
+
+        # Load the recommended.py file as a module
+        rec = imp.load_source('rec', filepath)
```

Comment:
Dynamically loading the module is a really good idea. The old code was so ugly... You might want to catch possible exceptions here and raise `DatabaseError`s, similarly to what the previous code did.

Reply:
Added